src/main/java/fr/zcraft/quartzlib/tools/items/ItemUtils.java

```diff
@@ -517,4 +520,88 @@ public static void dropLater(final Location location, final ItemStack item) {
         RunTask.nextTick(() -> drop(location, item));
     }
 
+    /**
+     * Converts a chat color to its dye equivalent.
+     *
+     * <p>The transformation is not perfect as there is no 1:1
+     * correspondence between dyes and chat colors.</p>
+     *
+     * @param color The chat color.
+     * @return The corresponding dye.
+     */
+    @Contract(pure = true)
+    public static DyeColor asDye(@NotNull final ChatColor color) {
+        switch (color) {
+            case BLACK:
+                return DyeColor.BLACK;
+
+            case BLUE:
+            case DARK_BLUE:
+                return DyeColor.BLUE;
+
+            case DARK_GREEN:
+                return DyeColor.GREEN;
+
+            case DARK_AQUA:
+                return DyeColor.CYAN;
+
+            case DARK_RED:
+                return DyeColor.RED;
+
+            case DARK_PURPLE:
+                return DyeColor.PURPLE;
+
+            case GOLD:
+            case YELLOW:
+                return DyeColor.YELLOW;
+
+            case GRAY:
+                return DyeColor.LIGHT_GRAY;
+
+            case DARK_GRAY:
+                return DyeColor.GRAY;
+
+            case GREEN:
+                return DyeColor.LIME;
+
+            case AQUA:
+                return DyeColor.LIGHT_BLUE;
+
+            case RED:
+                return DyeColor.ORANGE;
+
+            case LIGHT_PURPLE:
+                return DyeColor.PINK;
+
+            // White, reset & formatting
+            default:
+                return DyeColor.WHITE;
```

Comment:
Since conversion is not perfect, maybe this method should return an `Optional<DyeColor>` instead so that users can differenciate between white and the "reset & formatting" chat color codes?

Reply:
Why not, but what would be the behaviour of `colorize(ColorablleMaterial, ChatColor)`?

Fallback to white or `Optional<Material>`? I feel like a fallback would be way more usable

Reply:
I think a fallback to white is alright in this case, yes. If the user wants to customize the behavior, they can handle the `asDye` Optional with what they want, and then pass the verified result to `colorize(ColorableMaterial, DyeColor)`.

Reply:
Fine for me.